Update user views

lauschof opened the pull request:
User and Workshop views permissions added. Both tested in Insomnia. 

Class IsAdminOwnerOrSuperuser(permissions.BasePermission) was a bit tricky. Claude was kind enough to help, as the course content/online guides were not a much a help for creating different user levels. :D 

I left all the comments in to help understand what's happening! Message me if something doesnt make sense: :)

Commit:
stated making permission chnages

tscott-1 committed:
update from develop

Commit:
added user permissions

Commit:
edited workshop/note permissions

tscott-1 commented:
test: logged in as superuser
- can get users
- can update/archive a workshop i do not own
- 
test: logged in as staff 
- create workshop
- cannot update a workshop I do not own
- can update a workshop I own
- can add a note to my workshop
- can add a note to someone else's workshop
- 
test: logged in as user
- cannot create workshop
- cannot update a workshop
**- cannot add a note**

test: not logged in
- can see all workshops
- cannot add workshops
- cannot add a note

tscott-1 committed:
fix typo in view class

vewclass was set to NoteList, but the view is Notelist - fixed error to allow users to post!

tscott-1 committed:
add notedetail permissions list so regular users can edit their own notes

tscott-1 committed:
added permission check at get user so users cannot update other users

tscott-1 commented:
Fixed a typo
As a user/admin/superuser, I CAN post a note:)

as a user:
- I cannot archive another user's note
- i can get details of my note
**- I cannot archive my own note** ** fixed this by adding NoteDetail view to permissions list with Noteslist
- i cannot add a location
- i cannot  add an organisation
- i cannot get a user list
- i cannot get a users details
**- i CAN update another users details!!** - fixed moving some permission checks to the get

as an admin
- i cannot archive a note that is not on my workshop
- i can archive a note from another user on my workshop
- i cannot get notes detail for notes from another workshop
- i can get notes detail for notes on my workshops
- i cannot add a location
- i cannot add an organisation
- i cannot get a user list
- - i cannot get a users details


as a superuser
- i can get any notes details
- i can archive any note
- I can add a location
- i can add an organisation
- i can get a user list
- i can get a users details
- i can update users details

not logged in:
- i can get all notes
- i can't see any notes details
- i can't archive any notes
- I cannot add a location
- I can see all locations
- I can see all organisations
- i cannot add an organisation
- i can add a user
- i cannot get a user list
- i cannot get a users details
- i cannot update a users details

tscott-1 reviewed (approved):
fixed a few issues, good to go
